src/baseTrie.ts

```diff
@@ -389,16 +441,16 @@ module.exports = class Trie {
       stack.push(newBranchNode)
 
       if (lastKey.length !== 0) {
-        const branchKey = lastKey.shift()
+        const branchKey = lastKey.shift() as number
 
         if (lastKey.length !== 0 || lastNode.type === 'leaf') {
           // shriking extention or leaf
           lastNode.key = lastKey
           const formatedNode = this._formatNode(lastNode, false, toSave)
```

Comment:
Yeah I changed the signature because it was both confusing and didn't play nice with typescript. The reason was that out of the 4 arguments, the 3rd one was optional and defaulted to `false`. I moved the optional argument to be the last one. That's why this line didn't need to change. It wasn't a bug before.